pkgs/development/compilers/llvm/16/default.nix

```
@@ -306,6 +322,13 @@ in let
       bintools = bintools';
       extraPackages = [ ];
       extraBuildCommands = mkExtraBuildCommands0 cc;
+      nixSupport.cc-cflags = [
```

This breaks the Darwin stdenv bootstrap, which needs to use clangNoCompilerRtWithLibc to build Bash, Libsystem, CF, and their dependencies.

```suggestion
      nixSupport.cc-cflags = lib.optionals (!stdenv.hostPlatform.isDarwin) [
```

---

I found a possibly better solution, which is to pass `-nostdlib` only to libcxxabi (which it turns out the darwin bootstrap also used early on, so I removed it from there). This prevents clang from trying to link the startfiles and libgcc.